src/adtypes_interface.jl

```diff
@@ -133,18 +134,60 @@ julia> hessian_sparsity(f, [1.0, 2.0, 3.0, 4.0], TracerLocalSparsityDetector())
 """
 struct TracerLocalSparsityDetector{TG <: GradientTracer, TH <: HessianTracer} <:
     ADTypes.AbstractSparsityDetector end
-function TracerLocalSparsityDetector(
-        ::Type{TG}, ::Type{TH}
-    ) where {TG <: GradientTracer, TH <: HessianTracer}
-    return TracerLocalSparsityDetector{TG, TH}()
+
+#==========================#
+# Convenience constructors #
+#==========================#
+
+for D in (:TracerSparsityDetector, :TracerLocalSparsityDetector)
+    # Specity both tracer types as arguments
+    @eval ($D)(::Type{TG}, ::Type{TH}) where {TG <: GradientTracer, TH <: HessianTracer} = ($D){TG, TH}()
+
+    # Only specify one tracer type, fall back to default for other
+    @eval ($D)(::Type{TG}) where {TG <: GradientTracer} = ($D){TG, DEFAULT_HESSIAN_TRACER}()
+    @eval ($D)(::Type{TH}) where {TH <: HessianTracer} = ($D){DEFAULT_GRADIENT_TRACER, TH}()
+
+    # Convenience constructor: Only provide pattern types
+    # TODO: document
+    @eval function ($D)(;
+            gradient_pattern_type::Type{G} = DEFAULT_SET_TYPE,
+            hessian_pattern_type::Type{H} = DEFAULT_DICT_TYPE,
+            shared_hessian::Type{S} = DEFAULT_SHARED_TYPE,
+        ) where {G, H, S}
+        TG = gradient_tracer_type(G)
+        TH = hessian_tracer_type(H, S)
+        return ($D)(TG, TH)
+    end
```

Comment:
This will be the new public API for pattern selection.

Reply:
I've had issues in the past with Julia / JET failing to infer types from keyword arguments like those, even when they are properly annotated with `::Type{T} where T`. This is the reason why, in DI, the keyword `strict::Val{S}` is then passed as a positional argument to the chain of lower-level calls (and it's a pain in the butt).
Maybe it's not a problem here though, since the nesting is limited.